common/services/catalogue/codecs.js

```diff
@@ -0,0 +1,255 @@
+  link: (t: $Shape<T>) => NextLinkType,
+  query: (t: $Shape<T>) => RequestQuery,
+|};
+
+// /works
+export const WorksCodec: NextRoute<WorksParams> = {
+  fromQuery(q) {
+    return {
+      query: q.query || '',
+      page: q.page ? parseInt(q.page, 10) : null,
+      workType: stringTocsv(q.workType),
+      itemsLocationsLocationType: stringTocsv(
+        q['items.locations.locationType']
+      ),
+      sort: q.sort,
+      sortOrder: q.sortOrder,
+      productionDatesFrom: q['production.dates.from'],
+      productionDatesTo: q['production.dates.to'],
+      search: q.search,
+      source: q.source,
+    };
+  },
+
+  link(params) {
+    const pathname = '/works';
+
+    return {
+      href: {
+        pathname,
+        query: WorksCodec.query(params),
+      },
+      as: {
+        pathname,
+        query: WorksCodec.query({ ...params, source: null }),
+      },
+    };
+  },
+
+  query(params) {
+    return serialiseUrl({
+      query: params.query,
+      page: params.page,
+      workType: params.workType,
+      'items.locations.locationType': params.itemsLocationsLocationType,
+      sort: params.sort,
+      sortOrder: params.sortOrder,
+      'production.dates.from': params.productionDatesFrom,
+      'production.dates.to': params.productionDatesTo,
+      search: params.search,
+      source: params.source,
+    });
+  },
+};
+
+// /work/{id}
+export type WorkParams = {|
+  id: string,
+|};
+export const WorkCodec: NextRoute<WorkParams> = {
+  fromQuery(q) {
+    const { id } = q;
+    return { id };
+  },
+  link(params) {
+    const { id } = params;
+    return {
+      href: {
+        pathname: `/work`,
+        query: WorkCodec.query({
+          id,
+        }),
+      },
+      as: {
+        pathname: `/works/${id}`,
+      },
+    };
+  },
+  query(params) {
+    const { id } = params;
+    return { id };
+  },
+};
+
+// /works/{id}/items
+export type ItemParams = {|
+  workId: string,
+  langCode: string,
+  canvas: number,
+  sierraId: ?string,
+  isOverview?: boolean,
+  // This is used exclusively for
+  page: number,
+  pageSize: number,
+|};
+
+export const ItemCodec: NextRoute<ItemParams> = {
+  fromQuery(q) {
+    const {
+      workId,
+      langCode = 'eng',
+      canvas,
+      sierraId,
+      isOverview,
+      page,
+      pageSize,
+    } = q;
+    return {
+      workId,
+      langCode,
+      sierraId,
+      pageSize: pageSize ? parseInt(pageSize, 10) : 4,
+      canvas: defaultTo1(canvas),
+      isOverview: Boolean(isOverview),
+      page: defaultTo1(page),
+    };
+  },
+  link(params) {
+    const { workId, ...as } = params;
+    return {
+      href: {
+        pathname: `/item`,
+        query: ItemCodec.query(params),
+      },
+      as: {
+        pathname: `/works/${params.workId}/items`,
+        query: ItemCodec.query(as),
+      },
+    };
+  },
+  query(params) {
+    return serialiseUrl(params);
+  },
+};
+
+export const defaultWorkTypes = ['a', 'k', 'q', 'i', 'g'];
```

Comment:
```suggestion
export const defaultWorkTypes = ['a', 'g', 'i', 'k', 'l', 'q'];
```
we've added ephemera (l) to this